com.unity.render-pipelines.high-definition/Editor/AssetProcessors/MaterialPostProcessor.cs

```diff
@@ -371,6 +372,14 @@ static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAsse
                     // we would miss re-importing that dependency.
                     if (MaterialReimporter.CheckHDShaderGraphVersionsForUpgrade("", material.shader, ignoreNonHDRPShaderGraphs: false))
                     {
+                        s_ImportedMaterialCounter.TryGetValue(asset, out var importCounter);
+                        s_ImportedMaterialCounter[asset] = ++importCounter;
+
+                        // CheckHDShaderGraphVersionsForUpgrade always return true if a ShaderGraph don't have an HDMetaData attached
```

Comment:
question, can't we check that the shadergraph currently have or not the HDMetaData  data instead of relying on importCounter info?

Reply:
It's complicated because an HD ShaderGraph could not have an HDMetaData if it's old, in this case, the ShaderGraph needs to be reimported along with the material. It's only after this reimport that we can know if the material has or not an HDMetaData (i.e. the second time the material is imported).